Add OpenEdx E-Commerce app

jmaupetit opened the pull request:
## Purpose

We need to add OpenEdx E-Commerce application to our collection of Arnold's applications.

## Proposal

- [x] integrate first draft release of `edxec`

During this work, I have also noticed that the `lint-git` job was missing from our CI workflow. This is now fixed! :wink:

lunika committed:
✨(apps) add edxec application

We need to add OpenEdx E-Commerce application to our collection of
Arnold's applications. This is a first draft implementation of what is
required to run it with OpenShift.

jmaupetit commented on apps/edxec/templates/services/mysql/svc.yml.j2:
```diff
@@ -0,0 +1,27 @@
+    targetPort: {{ edxec_mysql_port }}
+# As commented in the ad hoc endpoint, the endpoint name points to this service
+# so that it does not rely on a deployment configuration when the "env_type" is
+# not trashable. In this case, we use a PostgreSQL cluster outside of
```
```suggestion
# not trashable. In this case, we use a MySQL cluster outside of
```

jmaupetit commented on apps/edxec/vars/vault/main.yml.j2:
```diff
@@ -0,0 +1,12 @@
+      PASSWORD: "{{ mysql_credentials.password }}"
+      HOST: "edxec-mysql-{{ deployment_stamp }}"
+      PORT: {{ edxec_mysql_port }}
+      ATOMIC_REQUESTS: true
```
This template should be refactored to fit with current required entries.

sampaccoud commented on apps/edxec/templates/services/app/job_00_collectstatic.yml.j2:
```diff
@@ -0,0 +1,79 @@
+          volumeMounts:
+            - name: edxec-local
+              mountPath: /usr/local/etc/ecommerce
+            - name: edxec-v-media
```
You shouldn't need to mount media to collectstatic?

jmaupetit replied:
Good point, thx!

sampaccoud commented on apps/edxec/templates/services/app/job_01_db_migrate.yml.j2:
```diff
@@ -0,0 +1,77 @@
+              mountPath: /usr/local/etc/ecommerce
+            - name: edxec-v-media
+              mountPath: /app/media
+            - name: edxec-v-static
```
You shouldn't need to mount statics to run db_migrate?

jmaupetit replied:
:pray:

sampaccoud commented on apps/edxec/vars/vault/main.yml.j2:
```diff
@@ -0,0 +1,16 @@
+# MySQL
+{% set mysql_credentials = databases.mysql | json_query("[?release=='" ~ edxapp_mysql_version ~ "'].databases | [0][?application=='edxec'].{user: user, password: password, name: name} | [0]") %}
```
```suggestion
{% set mysql_credentials = databases.mysql | json_query("[?release=='" ~ edxec_mysql_version ~ "'].databases | [0][?application=='edxec'].{user: user, password: password, name: name} | [0]") %}
```

lunika replied:
```suggestion
{% set mysql_credentials = databases.mysql | json_query("[?release=='" ~ edxec_mysql_version ~ "'].databases | [0][?application=='edxec'].{user: user, password: password, name: name} | [0]") %}
```

sampaccoud commented on apps/edxec/templates/services/mysql/secret.yml.j2:
```diff
@@ -0,0 +1,14 @@
+  name: "{{ edxec_mysql_secret_name }}"
+  namespace: "{{ project_name }}"
+data:
+  MYSQL_DATABASE: "{{ EDXEC_VAULT.MYSQL_DATABASE | default('edxapp') | b64encode }}"
```
```suggestion
  MYSQL_DATABASE: "{{ EDXEC_VAULT.MYSQL_DATABASE | default('edxec') | b64encode }}"
```

lunika replied:
```suggestion
  MYSQL_DATABASE: "{{ EDXEC_VAULT.MYSQL_DATABASE | default('edxec') | b64encode }}"
```

sampaccoud commented on apps/edxec/templates/services/mysql/secret.yml.j2:
```diff
@@ -0,0 +1,14 @@
+  namespace: "{{ project_name }}"
+data:
+  MYSQL_DATABASE: "{{ EDXEC_VAULT.MYSQL_DATABASE | default('edxapp') | b64encode }}"
+  MYSQL_USER: "{{ EDXEC_VAULT.MYSQL_USER | default('edxapp_user') | b64encode }}"
```
```suggestion
  MYSQL_USER: "{{ EDXEC_VAULT.MYSQL_USER | default('edxec_user') | b64encode }}"
```

lunika replied:
```suggestion
  MYSQL_USER: "{{ EDXEC_VAULT.MYSQL_USER | default('edxec_user') | b64encode }}"
```

lunika commented on apps/edxec/templates/services/app/secret.yml.j2:
```diff
@@ -0,0 +1,10 @@
+  name: "{{ edxec_secret_name }}"
+  namespace: "{{ project_name }}"
+data:
+  local.yml: "{{ EDXEC_VAULT.LOCAL_FILE | default('') | to_nice_yaml | b64encode }}"
```
Is it allow to have an empty local file ?

jmaupetit replied:
Yes.

lunika commented on apps/edxec/vars/all/main.yml:
```diff
@@ -0,0 +1,35 @@
+edxec_app_replicas: 1
+edxec_secret_name: "edxec-{{ edxec_vault_checksum | default('undefined_edxec_vault_checksum') }}"
+edxec_activate_http_basic_auth: false
+edxec_yq_image_name: "mikefarah/yq"
```
do you really think it is needed to set this image name in a variable ?

jmaupetit replied:
:man_shrugging: I wanted the tag to be configurable, so by extension, the image is also configurable. Do you think it's useless?

lunika replied:
I think yes but it's not a big deal

jmaupetit committed:
👷(circle) add lint-git job

For cross-projects consistency, we need to also check that our Git
commit messages fit with our guidelines.

jmaupetit commented:
@sampaccoud & @lunika I've addressed all your comments. Another review would be warmly appreciated :pray:

lunika commented on .circleci/config.yml:
```diff
@@ -92,6 +92,25 @@ jobs:
             - images
 
+  # Check that the git history is clean and complies with our expectations
+  lint-git:
```
![](https://media1.giphy.com/media/5GoVLqeAOo6PK/giphy.gif)

sampaccoud replied:
:medal_sports:

lunika commented on .circleci/config.yml:
```diff
@@ -362,6 +381,37 @@ jobs:
+      - *run_openshift_cluster
+
+      - run:
+          name: Test the "edxec" application bootstrapping (with redis enabled)
```
```suggestion
          name: Test the "edxec" application bootstrapping
```

madmatah commented on apps/edxec/templates/services/app/configs/local.yml.j2:
```diff
@@ -0,0 +1,49 @@
+    HOST: 'edxec-mysql-{{ deployment_stamp }}'
+    PORT: 3306
+    ATOMIC_REQUESTS: true
+# FIXME: is this really required?
```
This FIXME is about _ECOMMERCE_URL_ROOT_ ?

It seems to be mandatory, see [here](https://github.com/edx/ecommerce/blob/4b48dc739b42efdd746ede627bffdc7c4ad3d438/e2e/config.py#L20) and [here](https://github.com/edx/ecommerce/blob/4b48dc739b42efdd746ede627bffdc7c4ad3d438/e2e/helpers.py#L48)

And the default value set in the [docker image](https://github.com/openfun/openedx-ecommerce-docker/blob/cfbb22acf4d076fc86c52c5ee53efdecfc448995/docker/files/usr/local/etc/ecommerce/local.yaml#L11) is `http://localhost:8000`

jmaupetit replied:
Well spotted dude! This was a note to self but you did my homework :wink: This is a minor issue for us as the URL may change during the (blue/green) deployment. After checking the source code, it seems that it's only used for end-to-end testing. I'll remove my comment then.
